sensor/common/clusterentities/store_ips_test.go

```diff
@@ -0,0 +1,539 @@
+package clusterentities
+
+import (
+	"golang.org/x/exp/maps"
+
+	"github.com/stackrox/rox/pkg/net"
+	"github.com/stackrox/rox/pkg/networkgraph"
+)
+
+func (s *ClusterEntitiesStoreTestSuite) TestMemoryAboutPastIPs() {
+	cases := map[string]struct {
+		numTicksToRemember uint16
+		entityUpdates      map[int][]eUpdate // tick -> updates
+		operationAfterTick map[int]operation
+		endpointsAfterTick []map[string]whereThingIsStored
+		publicIPsAtCleanup []string
+	}{
+		"Memory disabled should forget 10.0.0.1 immediately": {
+			numTicksToRemember: 0,
+			entityUpdates: map[int][]eUpdate{
+				0: {
+					{
+						deploymentID: "depl1",
+						containerID:  "pod1",
+						ipAddr:       "10.0.0.1",
+						port:         80,
+						incremental:  true, // append
+					},
+					{
+						deploymentID: "depl1",
+						containerID:  "pod1",
+						ipAddr:       "10.3.0.1",
+						port:         80,
+						incremental:  false, // delete all data for the deployment, then apply update
+					},
+				},
+			},
+			endpointsAfterTick: []map[string]whereThingIsStored{
+				{"10.0.0.1": nowhere, "10.3.0.1": theMap}, // pre-tick 1: 10.0.0.1 should be overwritten immediately - only 10.3.0.1 should exist
+				{"10.0.0.1": nowhere, "10.3.0.1": theMap}, // tick 1: only 10.3.0.1 should exist
+				{"10.0.0.1": nowhere, "10.3.0.1": theMap}, // tick 2: only 10.3.0.1 should exist
+			},
+		},
+		"Old IPs should be gone on the first tick": {
+			numTicksToRemember: 1,
+			entityUpdates: map[int][]eUpdate{
+				0: {
+					{
+						deploymentID: "depl1",
+						containerID:  "pod1",
+						ipAddr:       "10.0.0.1",
+						port:         80,
+						incremental:  true,
+					},
+					{
+						deploymentID: "depl1",
+						containerID:  "pod1",
+						ipAddr:       "10.3.0.1",
+						port:         80,
+						incremental:  false,
+					},
+				},
+			},
+			endpointsAfterTick: []map[string]whereThingIsStored{
+				{"10.0.0.1": history, "10.3.0.1": theMap}, // pre-tick 1: both must exist
+				{"10.0.0.1": nowhere, "10.3.0.1": theMap}, // after-tick 1: only 10.3.0.1 should exist
+				{"10.0.0.1": nowhere, "10.3.0.1": theMap}, // after-tick 2: only 10.3.0.1 should exist
+			},
+		},
+		"Updates of the same IP should not expire": {
+			numTicksToRemember: 2,
+			entityUpdates: map[int][]eUpdate{
+				0: {
+					{
+						deploymentID: "depl1",
+						containerID:  "pod1",
+						ipAddr:       "10.0.0.1",
+						port:         80,
+						incremental:  false,
+					},
+				},
+				2: {
+					{
+						deploymentID: "depl1",
+						containerID:  "pod1",
+						ipAddr:       "10.0.0.1",
+						port:         80,
+						incremental:  false,
+					},
+				},
+			},
+			endpointsAfterTick: []map[string]whereThingIsStored{
+				{"10.0.0.1": theMap}, // tick 0: update0
+				{"10.0.0.1": theMap}, // tick 1: mark update0 as historical
+				{"10.0.0.1": theMap}, // tick 2: historical update0 exists; add again in update2
```

Comment:
```suggestion
				{"10.0.0.1": inBoth}, // tick 2: historical update0 exists; add again in update2
```
I think it should be `inBoth` here rather than `theMap`, if both the historical update0 exists and a new one is added at the same IP.

Reply:
This is the case number 1 from my comment: https://github.com/stackrox/stackrox/pull/13340#discussion_r1869922891 where the new data is exactly the same as the old one. In that case, we do not move one "copy" to history.

Reply:
Got it